src/simulation/sensors/camera/camera.h

```diff
@@ -52,11 +55,58 @@ class Camera: public SysModel {
     void addCosmicRayBurst(const cv::Mat&, cv::Mat &mDst, double);
     void applyFilters(cv::Mat &mSource, cv::Mat &mDst);
 
+    void setParentName(const std::string& cameraParentName);
+    std::string getParentName() const;
+    void setCameraOn();
+    void setCameraOff();
+    int isCameraOn() const;
+    void setCameraID(int cameraId);
+    int getCameraID() const;
+    void setResolution(const Eigen::Vector2i& cameraResolution);
+    Eigen::Vector2i getResolution() const;
+    void setImageCadence(const uint64_t& cameraRenderRate);
+    uint64_t getImageCadence() const;
+    void setFieldOfView(const Eigen::Vector2d& fov);
+    Eigen::Vector2d getFieldOfView() const;
+    void setCameraBodyFramePosition(const Eigen::Vector3d& cameraPosition_B);
+    Eigen::Vector3d getCameraBodyFramePosition() const;
+    void setBodyToCameraMrp(const Eigen::Vector3d& cameraMrp_CB);
+    Eigen::Vector3d getBodyToCameraMrp() const;
+    void setFocalLength (double cameraFocalLength);
+    double getFocalLength() const;
+    void setGaussianPointSpreadFunction(int cameraGaussianPointSpreadFunction);
+    int getGaussianPointSpreadFunction() const;
+    void setCosmicRayFrequency(double cameraCosmicRayFrequency);
+    double getCosmicRayFrequency() const;
+    void setReadNoise(double cameraReadNoise);
+    double getReadNoise() const;
+    void setSystemGain(double cameraGain);
+    double getSystemGain() const;
+    void setEnableStrayLight(bool cameraEnablesStrayLight);
+    bool getEnableStrayLight() const;
+
+private:
+    std::string parentSpacecraftName{};  //!< [-] Name of the parent body to which the camera should be attached
+    int cameraIsImaging{}; //!< [-] Is the camera currently taking images
```

Comment:
:small_red_triangle_down: This should probably be a `bool`.

Reply:
done